trin-core/src/portalnet/overlay_service.rs

```diff
@@ -545,30 +542,15 @@ impl OverlayService {
     /// Attempts to build a `Content` response for a `FindContent` request.
     fn handle_find_content(&self, request: FindContent) -> Result<Content, OverlayRequestError> {
         // Attempt to derive the content ID for the content key.
-        let content_id = match self.protocol {
-            ProtocolId::State => {
-                let state_content_key =
-                    StateContentKey::from_bytes(request.content_key.as_slice())?;
-                state_content_key.derive_content_id()?
-            }
-            ProtocolId::History => {
-                let history_content_key =
-                    HistoryContentKey::from_bytes(request.content_key.as_slice())?;
-                history_content_key.derive_content_id()?
-            }
-            _ => {
-                warn!(
-                    "Attempt to find undefined content for {:?} protocol",
-                    self.protocol
-                );
-                return Err(OverlayRequestError::InvalidRequest(format!(
-                    "Content undefined for {:?} protocol",
-                    self.protocol,
-                )));
+        let content_id = match (K::try_from)(request.content_key) {
```

Comment:
sigp's discv5 implementation uses the standard of a descriptive name with a `T` (for type) in front of it, e.g.: `TContentKey`. I think that this would improve the readability here.